datanode/sqlsubscribers/order.go

```diff
@@ -44,17 +51,45 @@ func NewOrder(store OrderStore) *Order {
 }
 
 func (os *Order) Types() []events.Type {
-	return []events.Type{events.OrderEvent}
+	return []events.Type{
+		events.OrderEvent,
+		events.ExpiredOrdersEvent,
+	}
 }
 
 func (os *Order) Push(ctx context.Context, evt events.Event) error {
-	return os.consume(evt.(OrderEvent), evt.Sequence())
+	switch evt.Type() {
+	case events.OrderEvent:
+		return os.consume(evt.(OrderEvent), evt.Sequence())
+	case events.ExpiredOrdersEvent:
+		return os.expired(ctx, evt.(ExpiredOrdersEvent), evt.Sequence())
+	}
+	return nil
 }
 
 func (os *Order) Flush(ctx context.Context) error {
 	return os.store.Flush(ctx)
 }
 
+func (os *Order) expired(ctx context.Context, eo ExpiredOrdersEvent, seqNum uint64) error {
+	orders, err := os.store.GetByMarketAndID(ctx, eo.MarketID(), eo.OrderIDs())
```

Comment:
I'm not sure I see the point of this change, I see you get reduced event numbers, but from the POV of the datanode that's not the bottleneck  (is it?  can we see the pprof), the bottle neck is the number of order updates we have to persist into the database, and having to first lookup the order before repersisting will not reduce the number of persisted rows and may even slow down order persistence processing which IS the bottle neck.   This change seems more harmful than beneficial, we should have a group discussion on it I think, may be I am missing something but I have significant concern about it.